File: data/src/main/java/net/yupol/transmissionremote/data/api/model/AddTorrentResult.java
package net.yupol.transmissionremote.data.api.model;

import com.squareup.moshi.Json;

public class AddTorrentResult {
    @Json(name = "torrent-added") public TorrentEntity torrentAdded;
    @Json(name = "torrent-duplicate") public TorrentEntity torrentDuplicate;
}

File: domain/src/main/java/net/yupol/transmissionremote/domain/model/AddTorrentResult.kt
package net.yupol.transmissionremote.domain.model

class AddTorrentResult private constructor(
        val success: Boolean = false,
        val duplicate: Boolean = false,
        val error: Boolean = false,
        val errorMessage: String? = null)
{
    companion object {
        fun success() = AddTorrentResult(success = true)
        fun duplicate() = AddTorrentResult(duplicate = true)
        fun error(message: String? = null) = AddTorrentResult(error = true, errorMessage = message)
    }
}
